R/get_taxon_observations.R

```
@@ -1,39 +1,41 @@
 #' Get a single taxon observation
 #'
-#' Return a single taxon observation record from VegBank, using the
-#' observation's accession code.
+#' Return a single taxon observation record from VegBank, using
+#' its `to` code.
 #'
-#' @param accession_code A taxon observation accession code
+#' @param taxon_obs_code A taxon observation code
+#' @param parquet Transfer data as Parquet (TRUE) or JSON (FALSE)?
 #' @return A data frame (one row), or an empty data frame if there is
-#' no matching record for the provided accession code
+#' no matching record for the provided code
 #' @examples \dontrun{
-#' get_taxon_observation("VB.TO.64992.VACCINIUMBOREAL")
+#' get_taxon_observation("to.64992")
 #' }
 #' @import httr2
 #' @export
-get_taxon_observation <- function(accession_code) {
+get_taxon_observation <- function(taxon_obs_code, parquet = FALSE) {
   resource <- "taxon-observations"
-  get_resource_by_code(resource, accession_code)
+  get_resource_by_code(resource, taxon_obs_code, parquet = parquet)
 }
 
 #' Get all plot observations
 #'
 #' Return a paginated set of taxon observation records from VegBank.
 #'
-#' @return A data frame
 #' @param max_taxa_per_plot Max number of taxon observations to return
 #' per plot observation, in descending order of reported cover
 #' @param limit Number of records to return
 #' @param offset Number of records to skip
-#' @param detail Desired level of detail ("minimal", "full")
+#' @param parquet Transfer data as Parquet (TRUE) or JSON (FALSE)?
+#' @return A data frame
 #' @examples \dontrun{
 #' get_all_taxon_observations()
 #' }
 #' @import httr2
 #' @export
-get_all_taxon_observations <- function(max_taxa_per_plot = 5, limit=100, offset=0,
-    detail = c("minimal", "full")) {
```

Althought its not strictly under the objective of this PR, the `taxon-observations` endpoint doesn't support a "minimal detail" option, so I've removed it here. Previously, it silently returned "full detail" either way, but this turned into an error with the latest API changes.